pinot-core/src/main/java/org/apache/pinot/core/segment/index/converter/SegmentV1V2ToV3FormatConverter.java

```diff
@@ -85,7 +85,10 @@ public void convert(File v2SegmentDirectory)
 
     File newLocation = SegmentDirectoryPaths.segmentDirectoryFor(v2SegmentDirectory, SegmentVersion.v3);
     LOGGER.info("v3 segment location for segment: {} is {}", v2Metadata.getName(), newLocation);
-    v3TempDirectory.renameTo(newLocation);
+
+    if (!v3TempDirectory.renameTo(newLocation)) {
+      LOGGER.error("cannot rename {} to {}", v3TempDirectory.getName(), newLocation.getName());
```

Comment:
Hi, @mcvsubbu , thank you for review. I think you are right, it looks more appropriate to throw Exception here. Since it is not related with this pr(Windows compatible), I revert it back here, will submit a hotfix if necessary after.